weave/weave_client.py

```diff
@@ -739,8 +739,11 @@ def _save_op(self, op: Op, name: Optional[str] = None) -> Ref:
 
         if inspect.ismethod(op):
             # this "func" is both a method AND an Op, but we need to cast to get dot access to its attributes
-            op = cast(op, Op)  # type: ignore
-        op.ref = op_def_ref  # type: ignore
+            op = cast(Op, op)
```

Comment:
this was flipped, which resulted in the ref not actually getting assigned to the object!

Reply:
Wow, great catch!